ANMN/burst_averaged_product/burst_average.py

```diff
@@ -0,0 +1,397 @@
+            var_max_burst    .append(np.max(var_values[valid_index_burst_range]))
+            var_sd_burst     .append(np.std(var_values[valid_index_burst_range]))
+            var_num_obs_burst.append(len(valid_index_burst_range))
+        else:
+            var_mean_burst   .append(fill_value)
+            var_min_burst    .append(fill_value)
+            var_max_burst    .append(fill_value)
+            var_sd_burst     .append(fill_value)
+            var_num_obs_burst.append(0)
+
+    burst_var = {'time_mean':   time_mean_burst,
+                 'var_mean':    var_mean_burst,
+                 'var_min':     var_min_burst,
+                 'var_max':     var_max_burst,
+                 'var_sd':      var_max_burst,
+                 'var_num_obs': var_num_obs_burst}
+
+    return burst_var
+
+def generate_netcdf_burst_filename(netcdf_file_obj, burst_vars):
+    """
+    generate the filename of a burst average netcdf for both CTD and WQM
+    """
+    site_code         = netcdf_file_obj.site_code
+    input_netcdf_name = os.path.basename(input_netcdf_file_path)
+    pattern           = re.compile("^(IMOS_.*)_([0-9]{8}T[0-9]{6}Z)_(.*)_(FV0[0-9])_(.*)_END")
+    match_group       = pattern.match(input_netcdf_name)
+
+    time_burst_vals = burst_vars['TEMP']['time_mean']
+    time_min        = num2date(time_burst_vals, netcdf_file_obj['TIME'].units, netcdf_file_obj['TIME'].calendar).min().strftime('%Y%m%dT%H%M%SZ')
+    time_max        = num2date(time_burst_vals, netcdf_file_obj['TIME'].units, netcdf_file_obj['TIME'].calendar).max().strftime('%Y%m%dT%H%M%SZ')
+    burst_filename  = "%s_%s_%s_FV02_%s-burst-averaged_END-%s_C-%s.nc" % (match_group.group(1), time_min, \
+                                                                          site_code, match_group.group(5), \
+                                                                          time_max, datetime.now().strftime("%Y%m%dT%H%M%SZ"))
+
+    return burst_filename
+
+def create_burst_average_netcdf(input_netcdf_file_path, output_dir):
+    """
+    generate the burst netcdf file for WQM product.
+    see variable conf_file if editing of gatt and var att need to be done
+    """
+    input_netcdf_obj = Dataset(input_netcdf_file_path, 'r')
+    burst_vars       = create_burst_average_var(input_netcdf_obj)
+    time_burst_vals  = burst_vars['TEMP']['time_mean']
+    tmp_netcdf_dir   = tempfile.mkdtemp()
+
+    output_netcdf_file_path = os.path.join(tmp_netcdf_dir, generate_netcdf_burst_filename(input_netcdf_obj, burst_vars))
+    output_netcdf_obj       = Dataset(output_netcdf_file_path, "w", format="NETCDF4")
+
+    # read gatts from input, add them to output. Some gatts will be
+    # overwritten
+    input_gatts     = input_netcdf_obj.__dict__.keys()
+    gatt_to_dispose = ['author', 'toolbox_input_file', 'file_version_quality_control', 'quality_control_set',
+                       'history', 'CoordSysBuilder_', 'netcdf_filename', 'metadata', 'featureType']
```

Comment:
@lbesnard when adding featureType you need to declare a TIMESERIES scalar variable with a specific role attribute. See WQM files processed by a recent toolbox > 2.4. Ex.: http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSMAI/Biogeochem_timeseries/IMOS_ANMN-NRS_KOSTUZ_20151206T185854Z_NRSMAI-SubSurface_FV01_NRSMAI-SubSurface-24-2015-11-WQM-85_END-20160512T035944Z_C-20160523T032107Z.nc.html